src/Microsoft.SqlTools.SqlCore/Connection/SqlRetryProvider.cs

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.SqlCore.Connection.ReliableConnection
+{
+    public static class SqlRetryProviders
```

Comment:
It would be great not to duplicate this...

It seems like having a small connection library that is used by this, STS and ManagedBatchParser wouldn't be a bad idea (and move all the "reliable connection" stuff from ManagedBatchParser to that). But something like that should be a separate PR - so maybe for this at least just move this class over completely and have ManagedBatchParser reference this instead?

@cheenamalhotra